Support compilation with GOARCH=wasm GOOS=wasip1

I was trying to compile Hugo into a WebAssembly WASIp1 executable and got a few errors stemming from this package:

```
🌳 hugo (master =) $ GOARCH=wasm GOOS=wasip1 go build -o hugo.wasm
# github.com/bep/mclib/internal
/home/ansemjo/.local/go/pkg/mod/github.com/bep/mclib@v1.20400.20402/internal/main.go:193:39: undefined: CertutilInstallHelp
/home/ansemjo/.local/go/pkg/mod/github.com/bep/mclib@v1.20400.20402/internal/main.go:195:77: undefined: NSSBrowsers
/home/ansemjo/.local/go/pkg/mod/github.com/bep/mclib@v1.20400.20402/internal/main.go:272:9: m.installPlatform undefined (type *mkcert has no field or method installPlatform)
...
```

Instead of commenting out all the uses of `mclib` in `commands/server.go`, the cleaner approach is probably to just make `mclib` compile to WebAssembly. The fix was pretty simple, I just added `wasm` to the `//go:build ...` line in `internal/truststore_other.go`. (Is it perhaps better to use a negation like `!(darwin || linux || windows)` right away? I haven't tested this.)

```diff
🌱 mclib ((v1.20400.20402) *) $ git diff
-----------------------------------------------
modified: internal/truststore_other.go
-----------------------------------------------
@ internal/truststore_other.go:1 @
-//go:build dragonfly || freebsd || openbsd || netbsd || solaris
+//go:build dragonfly || freebsd || openbsd || netbsd || solaris || wasm
 
 package internal
 
```

Then I used the local directory as a package replacement in Hugo to successfully compile a WASIp1 binary pretty much directly from `master`:

```
🌵 hugo (master =) $ go mod edit -replace github.com/bep/mclib=/tmp/tmpdir-qsfC1P/mclib
🌴 hugo (master *=) $ GOARCH=wasm GOOS=wasip1 go build -o hugo.wasm
🍃 hugo (master *%=) $ wasmtime run hugo.wasm version
hugo v0.156.0-DEV-e2776593c01f0f625cfed78093696ccb51535600 wasip1/wasm BuildDate=2026-02-11T12:34:10Z
```

Wether you want to run this binary in the browser (https://github.com/gohugoio/hugo/issues/8972) or just have a portable version of Hugo for archival / vendoring ... since the fix is so simple, I believe we could just make Hugo support this target easily. :)

And yes, you **can** actually build a website with this executable:

```
🌴 homepage (main *%>) $ wasmtime run --dir /tmp::/tmp --dir .::/homepage /tmp/tmpdir-CNjxWJ/hugo/hugo.wasm -s /homepage --noBuildLock build
Start building sites … 
hugo v0.156.0-DEV-e2776593c01f0f625cfed78093696ccb51535600 wasip1/wasm BuildDate=2026-02-11T12:34:10Z

WARN  Hint shortcode is deprecated, use markdown alerts format: '> [!NOTE] ...'

                  │ EN  
──────────────────┼─────
 Pages            │ 119 
 Paginator pages  │   3 
 Non-page files   │  64 
 Static files     │  71 
 Processed images │   0 
 Aliases          │  19 
 Cleaned          │   0 

Total in 2850 ms
```

--- a/gen/main.go
+++ b/gen/main.go
@@ -33,6 +33,10 @@ func main() {
 		log.Fatal(err)
 	}
 
+	if err := filehelpers.CopyFile("./staticcheck.conf", filepath.Join(internalDir, "staticcheck.conf")); err != nil {
+		log.Fatal(err)
+	}
+
 	fileReplacer := strings.NewReplacer(
 		"getCAROOT()", "GetCAROOT()",
 		"mkcert -install", "hugo server trust",

--- a/gen/main.go
+++ b/gen/main.go
@@ -5,6 +5,7 @@ package main
 import (
 	"log"
 	"os"
+	"os/exec"
 	"path/filepath"
 	"regexp"
 	"strings"
@@ -90,6 +91,13 @@ func main() {
 		return nil
 	})
 	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Fix missing/unused imports after the replacements above.
+	cmd := exec.Command("goimports", "-w", internalDir)
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
 		log.Fatal(err)
 	}
 }

--- a/gen/staticcheck.conf
+++ b/gen/staticcheck.conf
@@ -1,0 +1,1 @@
+checks = ["none"]

--- a/gen/truststore_other.go.txt
+++ b/gen/truststore_other.go.txt
@@ -1,4 +1,4 @@
-//go:build dragonfly || freebsd || openbsd || netbsd || solaris
+//go:build dragonfly || freebsd || openbsd || netbsd || solaris || wasm
 
 package internal
 

--- a/internal/truststore_darwin.go
+++ b/internal/truststore_darwin.go
@@ -22,7 +22,7 @@ var (
 )
 
 // https://github.com/golang/go/issues/24652#issuecomment-399826583
-var trustSettings []any
+var trustSettings []interface{}
 var _, _ = plist.Unmarshal(trustSettingsData, &trustSettings)
 var trustSettingsData = []byte(`
 <array>
@@ -67,7 +67,7 @@ func (m *mkcert) installPlatform() bool {
 
 	plistData, err := ioutil.ReadFile(plistFile.Name())
 	fatalIfErr(err, "failed to read trust settings")
-	var plistRoot map[string]any
+	var plistRoot map[string]interface{}
 	_, err = plist.Unmarshal(plistData, &plistRoot)
 	fatalIfErr(err, "failed to parse trust settings")
 
@@ -76,9 +76,9 @@ func (m *mkcert) installPlatform() bool {
 	if plistRoot["trustVersion"].(uint64) != 1 {
 		panic(fmt.Sprintln("ERROR: unsupported trust settings version:", plistRoot["trustVersion"]))
 	}
-	trustList := plistRoot["trustList"].(map[string]any)
+	trustList := plistRoot["trustList"].(map[string]interface{})
 	for key := range trustList {
-		entry := trustList[key].(map[string]any)
+		entry := trustList[key].(map[string]interface{})
 		if _, ok := entry["issuerName"]; !ok {
 			continue
 		}

--- a/internal/truststore_other.go
+++ b/internal/truststore_other.go
@@ -1,4 +1,4 @@
-//go:build dragonfly || freebsd || openbsd || netbsd || solaris
+//go:build dragonfly || freebsd || openbsd || netbsd || solaris || wasm
 
 package internal
 

--- a/internal/truststore_other.go
+++ b/internal/truststore_other.go
@@ -1,4 +1,4 @@
-//go:build dragonfly || freebsd || openbsd || netbsd || solaris || wasm
+//go:build dragonfly || freebsd || openbsd || netbsd || solaris
 
 package internal
 

